src/eradiate/scenes/shapes/_sphere.py

```diff
@@ -50,37 +53,20 @@ class SphereShape(ShapeNode):
         default="1.0",
     )
 
-    to_world: "mitsuba.ScalarTransform4f" = documented(
-        attrs.field(
-            converter=converters.to_mi_scalar_transform,
-            default=None,
-        ),
-        doc="Transform to scale, shift and rotate the sphere. "
-        "This transform will be prepended to the transform derived "
-        "from the center position and radius of the sphere. If, for example, a "
-        "scaling is provided here, it will multiply the sphere's radius.",
-        type="mitsuba.ScalarTransform4f or None",
-        init_type="mitsuba.ScalarTransform4f or array-like, optional",
-        default=None,
-    )
-
-    @to_world.validator
-    def to_world_validator(self, attribute, value):
-        if value is not None:
-            if not isinstance(value, mi.ScalarTransform4f):
-                raise TypeError(
-                    f"while validating '{attribute.name}': "
-                    f"'{attribute.name}' must be a mitsuba.ScalarTransform4f; "
-                    f"found: {type(value)}",
-                )
-
     @property
     def bbox(self) -> BoundingBox:
         length_units = ucc.get("length")
-        offset = (
-            np.full_like(self.center, self.radius.m_as(length_units)) * length_units
-        )
-        return BoundingBox(self.center - offset, self.center + offset)
+        offset = np.full_like(self.center, self.radius.m_as(length_units))
+        p1 = self.center.m_as(length_units) - offset
+        p2 = self.center.m_as(length_units) + offset
+
+        if self.to_world is not None:
+            p1 = self.to_world @ p1
+            p1 = pint.Quantity(np.array(p1), length_units)
+            p2 = self.to_world @ p2
+            p2 = pint.Quantity(np.array(p2), length_units)
```

Comment:
Not quite sure how this will work out if you apply anything else than a translation. Notably, if there is a rotation in the mix, you will rotate AABB points, while it should not happen (the Mitsuba plugin itself forbids nonuniform scalings and shears, so those will crash in a not-so-nice way but at least no weird stuff will happen).

IMO, the simplest is to recalculate the new centre and radius, and use the same formula as for the no-transform case.

Side comment: You are supposed to instantiate `Quantity` from the registry-specify type, not from the global Pint type. Please prefer the safer syntax:
```suggestion
            p1 = (np.array(self.to_world @ p1)) * length_units
            p2 = (np.array(self.to_world @ p2)) * length_units
```

Reply:
I am not sure, how to compute the new radius. Mitsuba transforms don't handle scalars. The center point is clear.